components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1115,6 +1119,11 @@ void RewardsServiceImpl::GetPublisherActivityFromUrl(uint64_t windowId, const st
       GetPublisherYear(now));
 }
 
+void RewardsServiceImpl::OnExcludedSitesChanged() {
+  for (auto& observer : observers_)
```

Comment:
we don't need observer here. We should get num directly from `OnExcludedSitesChanged`. This way you don't need to call native ledger again